core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts

```diff
@@ -186,11 +182,41 @@ api_doc_name:string='';
           )
   };
 
+  setTabs() {
+    if (this.service.serviceType === 'api' || this.service.type === 'api') {
+      this.tabData = ['overview', 'deployments', 'assets', 'code quality', 'logs'];
+    } else if (this.service.serviceType === 'function' || this.service.type === 'function') {
+      this.tabData = ['overview', 'deployments', 'assets', 'code quality', 'logs'];
+    } else if (this.service.serviceType === 'website' || this.service.type === 'website') {
+      this.tabData = ['overview', 'deployments', 'assets'];
+    }
+  }
+
+  getAssets() {
+    this.http.post('/jazz/assets/search', {
+      service: this.service.service || this.service.name,
+      domain: this.service.domain,
+      environment: this.envSelected,
+      limit: undefined
+    }).subscribe((assetsResponse) => {
+      this.assets = assetsResponse.data;
+      this.service.assets = this.assets;
+    }, (err) => {
+      this.toast_pop('error', 'Oops!', 'Swagger File Not Found.');
+    });
+  }
+
   testApi(type){
       switch(type){
           case 'api':
-            let swaggerFile = '/' + this.service.domain + '/' + this.service.name + '/' + this.envSelected + '/swagger.json';
-            return window.open(environment.urls['swagger_editor'] + '/?url=' + environment['api_doc_name'] + swaggerFile);
+            let foundAsset = this.assets.find((asset) => {
+              return asset.asset_type === 'swagger_url';
+            });
+            if (foundAsset) {
+              return window.open(environment.urls['swagger_editor'] + '/?url=' + foundAsset.provider_id);
```

Comment:
We don't want to hard code `?url=` so we have now included that as part of the key's value: https://github.com/tmobile/jazz-installer/pull/276/files#diff-04f485a5f936d3ceecd19be0e841d392R56.

Also we have slightly changed the key name to reflect its value, so reference it accordingly.